pages/heracles_explained.md

```diff
@@ -3,101 +3,203 @@ layout: page
-
-For Classes, `IClass` simply have `displayname`, `description`, `supportedOperation` and `supportedProperty` as attributes. For `supportedOperation` there is an interface `IOperation` which returns methods for operations; for `supportedProperty` there is an interface `ISupportedProperty` which have properties like `writeable`, `readable`. While properties can contain one more interface in them: `IProperty` which able to return display name, description and other properties. 
-
-
-### Design Patters of Client 
-
-#### Design for HydraClient Class:
-
-HydraClient is using for getting `Url`, `ApiDocumentationUrl`, and after that `ApiDocumentation` and also to check if an entrypoint exists or not. It simply works with the `IHyadraClient` interface which working to obtain `ApiDocumentation`, representation of `Resource`.
-
-HydraClient methods(simply functions):
-* `getApiDocumentation`
-* `getApiDocumentationUrl`
-* `getHypermediaProcessor`
-* `getResource`
-* `getUrl`
-
-#### Design for ApiDocumentation Class:
-
-`ApiDocumentation` is used to assign values to properties or variables like `type`, `title`, `description`, `supportedClasses`, `Entrypoint` etc. `ApiDocumentation` is the default implementation of `IApiDocumentation`.
-
-`ApiDocumentation` methods and properties:
-* `getEntrypoint`
-* `type`
-* `title`
-* `description`
-* `supportedClass`
-* `entrypoint`
-
-#### Design for JsonLdHypermediaProcessor Class:
-
-`JsonLdHypermediaProcessor` is a JSON-LD based implementation of `IHypermediaProcessor`.
-
-`JsonLdHypermediaProcessor` methods and properties:
-* `supportedMediaTypes`
-* `process`
-others are private methods.
-
-#### Design for ResourceEnrichmentProvider Class:
-
-`ResourceEnrichmentProvider` provides `IWebResource` enrichment routines. It enriches a given resource with `IHypermediaContainer` specific properties with method `enrichHypermedia`.
-
-`ResourceEnrichmentProvider` methods:
-* `enrichHypermedia`
-
-Actually, there are many interfaces which have track on the objects as well as its properties as we have disussed above and with the implementation of the interface (called Class above) means accessing the members or properties of interface for whom  client can get a lot of knowledge about the starting url or server.
-
-Load on server and speed can be limitation for the client. But in [python-hydra-agent](https://github.com/HTTP-APIs/python-hydra-agent)  we are dicsussing mainly about how to reduce the load on server and fast querying from the client. For this type of problem, we are using Redis to reduce load of server and we can implement indexing (specially secondary/faceted indexing) of objects and their properties in Redis with a good querying mechanism which can make operations faster and efficient.
+To better understand things, let’s do it by using Heracles.ts. We can use Heracles.ts in the browser as well as in the node environment.
```

Comment:
We should try to keep the docs in the third person as much as possible. Use something like "Heracles.ts can be installed ..."

Reply:
The same goes for the whole page